<grobian@gentoo.org>: perform endian check at runtime, as that is far
                      more portable than anything else without using
                      autoconf, the <endian.h> include is Linux only

--- PDFCore.cc.orig	2009-10-03 23:18:35 +0200
+++ PDFCore.cc	2009-10-04 10:49:40 +0200
@@ -16,7 +16,6 @@
 #include <locale.h> 
 #include <langinfo.h> 
 #include <iconv.h> 
-#include <endian.h> 
 #include "GooString.h"
 #include "GooList.h"
 #include "GlobalParamsGUI.h"
@@ -1617,11 +1616,12 @@
   
   nchars = len;
   if (len > 0) {
-#if __BYTE_ORDER == __LITTLE_ENDIAN 
-    cd = iconv_open ("UCS-4LE",nl_langinfo (CODESET)); 
-#else 
-    cd = iconv_open ("UCS-4BE",nl_langinfo (CODESET)); 
-#endif 
+    long one = 1;
+    if (*((char *)(&one))) {
+      cd = iconv_open ("UCS-4LE",nl_langinfo (CODESET)); 
+    } else {
+      cd = iconv_open ("UCS-4BE",nl_langinfo (CODESET)); 
+    }
     inptr = s; 
     outptr = (char *) u; 
     insize = len; 
